tests/instruments/ipgphotonics/test_yar.py

```diff
@@ -0,0 +1,233 @@
+    with expected_protocol(
+            YAR,
+            init_comm + [("ROP", "ROP: Off")]
+    ) as inst:
+        assert inst.power == 0
+
+
+def test_power_low():
+    """Verify, that power 'Low' is translated to the default minimum 0.1."""
+    with expected_protocol(
+            YAR,
+            init_comm + [("ROP", "ROP: Low")]
+    ) as inst:
+        assert inst.power == 0.1
+
+
+def test_init_sets_minimum_display_power():
+    """Test, that a different minimum power is shown in the translation of 'Low'."""
+    # not in the manual!
+    with expected_protocol(
+            YAR,
+            # init_comm is modified
+            [("RNP", "RNP: 0.200"), ("RMP", "RMP: 10.5"), ("RDPT", "RDPT: 5"),
+             ("ROP", "ROP: Low")]
+    ) as inst:
+        assert inst.power == 5
+
+
+def test_power_setpoint_getter():
+    with expected_protocol(
+            YAR,
+            init_comm + [("RPS", "RPS: 1.23")]
+    ) as inst:
+        assert inst.power_setpoint == 1.23
+
+
+def test_power_setpoint_setter():
+    with expected_protocol(
+            YAR,
+            init_comm + [("SPS 9.7", "SPS: 9.7")]
+    ) as inst:
+        inst.power_setpoint = 9.7
+
+
+def test_power_setpoint_setter_out_of_range_driver():
+    with expected_protocol(
+            YAR,
+            # init_comm is modified
+            [("RNP", "RNP: 0.200"), ("RMP", "RMP: 10.5"), ("RDPT", "RDPT: 0.100")]
+    ) as inst:
+        with pytest.raises(ValueError):
+            inst.power_setpoint = 20
+
+
+def test_power_setpoint_setter_out_of_range_device():
+    with expected_protocol(
+            YAR,
+            init_comm + [("SPS 5", "ERR: Out of Range")]
+    ) as inst:
+        with pytest.raises(ConnectionError):
+            inst.power_setpoint = 5
+
+
+def test_current():
+    with expected_protocol(
+            YAR,
+            init_comm + [("RDC", "RDC: 2.34")]
+    ) as inst:
+        assert inst.current == 2.34
+
+
+def test_temperature():
+    with expected_protocol(
+            YAR,
+            init_comm + [("RCT", "RCT: 28.9")]
+    ) as inst:
+        assert inst.temperature == 28.9
+
+
+def test_wavelength_temperature_getter():
+    with expected_protocol(
+            YAR,
+            init_comm + [("RWA", "RWA: 31.345")]
+    ) as inst:
+        assert inst.wavelength_temperature == 31.345
+
+
+def test_wavelength_temperature_setter():
+    with expected_protocol(
+            YAR,
+            init_comm + [("SWA 35.658", "SWA: 35.658")]
+    ) as inst:
+        inst.wavelength_temperature = 35.658
+
+
+def test_seed_temperature():
+    with expected_protocol(
+            YAR,
+            init_comm + [("RST", "RST: 31.345")]
+    ) as inst:
+        assert inst.temperature_seed == 31.345
+
+
+def test_clear():
+    with expected_protocol(
+            YAR,
+            init_comm + [("RERR", "RERR")]
+    ) as inst:
+        inst.clear()
+
+
+def test_model():
+    """This test tests already getting min/max output power."""
```

Comment:
docstring of this and the next test probably copy/pasted?

Reply:
Yes, Done.